internal/pkg/renderer/renderer.go

```diff
@@ -164,13 +173,20 @@ func prepareTemplates(p *pack.Pack, templates map[string]toRender, variables map
 	}
 	// Iterate the dependencies and prepareTemplates for each.
 	for _, child := range p.Dependencies() {
-		prepareTemplates(child, templates, newVars)
+		prepareFiles(child, templates, auxFiles, newVars, renderAuxFiles)
 	}
 
 	// Add each template within the pack with scoped variables.
 	for _, t := range p.TemplateFiles {
 		templates[path.Join(p.Name(), t.Name)] = toRender{content: string(t.Content), variables: newVars}
 	}
+
+	if renderAuxFiles {
+		// Add each aux file within the pack with scoped variables.
+		for _, f := range p.AuxiliaryFiles {
+			templates[path.Join(p.Name(), f.Name)] = toRender{content: string(f.Content), variables: newVars}
```

Comment:
```suggestion
			auxFiles[path.Join(p.Name(), f.Name)] = toRender{content: string(f.Content), variables: newVars}
```

I think? Otherwise `auxFiles` doesn't seem to be used anywhere. But then I think `Renderer.Render` needs to handle this map, which it doesn't seem to be doing.

More general, it may be worth refactoring this function so it doesn't take maps as inputs, but rather return them.

Reply:
Addressed in 6151776461dc689ac79b6dd45011e88b8103c912